### PR TITLE
Fix #275545: duplicated property tags

### PR DESCRIPTION
**Pull request by jthistle**
References: https://musescore.org/en/node/275545

### CODE REVIEW

**Comment by lasconic** [comment]
I wonder if this is the right thing to do. It seems it will change the current behavior. Frame shape is not a good example but what if a property has 3 different value like 0 1 and 2 with 0 the default. If the element has 1 and the style has 2 then currently we write both 1 and 2. When we read the file, 2 is considered since it's the second one. If we applied this patch , 1 will be used instead. Is it what we want ? No idea :) What do you think @wschweer @anatoly-os  ?

**Comment by jthistle** [comment]
Oh, that's interesting. So the duplicate tags isn't really a bug, but it's meant to be like that?

**Comment by lasconic** [comment]
No, I think the duplicated tag is a bug but what I mean is that your code will print the first tag only, while the current reading code (I believe) would consider the second tag instead.

**Comment by jthistle** [comment]
oh, so if I change the code to just overwrite the property with the same pid in writtenProperties instead of skipping it, then it should work as you're describing?

**Comment by jthistle** [comment]
Latest commit fixes merge conflicts.

File: libmscore/textbase.cpp [hunk 1]
```diff
@@ -1675,14 +1675,26 @@ void TextBase::writeProperties(XmlWriter& xml, bool writeText, bool /*writeStyle
...
+                  continue
+            if (writtenProperties.indexOf(spp.pid) != -1)
+                  continue;
+            writeProperty(xml, spp.pid);
+            writtenProperties.append(spp.pid);
             }
       for (const StyledProperty& spp : *textStyle(tid())) {
             if (!isStyled(spp.pid))
-                  writeProperty(xml, spp.pid);
+                  continue
```
Comment (Jojo-Schmitz) [inline libmscore/textbase.cpp]: semicolon `;` missing

File: libmscore/textbase.cpp [hunk 2]
```diff
@@ -1675,14 +1675,26 @@ void TextBase::writeProperties(XmlWriter& xml, bool writeText, bool /*writeStyle
...
+            if (writtenProperties.indexOf(spp.pid) != -1)
```
Comment (Jojo-Schmitz) [inline libmscore/textbase.cpp]: why not `if (!isStyled(spp.pid) || writtenProperties.indexOf(spp.pid) != -1)`?

File: libmscore/textbase.cpp [hunk 3]
```diff
@@ -1675,14 +1675,26 @@ void TextBase::writeProperties(XmlWriter& xml, bool writeText, bool /*writeStyle
...
       writeProperty(xml, Pid::SUB_STYLE);
 
+      // used to prevent duplicate writing of any property
+      QVector<Pid> writtenProperties;
+
       for (const StyledProperty& spp : *_elementStyle) {
             if (!isStyled(spp.pid))
-                  writeProperty(xml, spp.pid);
+                  continue
+            if (writtenProperties.indexOf(spp.pid) != -1)
```
Comment (Jojo-Schmitz) [inline libmscore/textbase.cpp]: Why not `if (!isStyled(spp.pid) || writtenProperties.indexOf(spp.pid) != -1)`?

**Comment by Jojo-Schmitz** [comment]
But (now?) it  fails on Travis CI and AppVeyor, and for very good reason: the code doesn't compile

**Comment by jthistle** [comment]
ugh I rushed this and didn't build it before I pushed, sorry. Will fix and update tommorow.

**Comment by Jojo-Schmitz** [comment]
Now some mtests fail, pretty clearly related to this PR

**Comment by jthistle** [comment]
Closing, imo this fix isn't sufficient, and the issue may also not exist.

**Pull request closed without merging**